components/data/data-csvim/src/main/java/org/eclipse/dirigible/components/data/csvim/synchronizer/CsvimProcessor.java

```diff
@@ -156,7 +156,7 @@ public void process(CsvFile csvFile, InputStream content, Connection connection)
         TableMetadata tableMetadata = CsvimUtils.getTableMetadata(tableName, connection);
         String pkName = getPkName(tableMetadata, csvParser.getHeaderNames());
 
-        if (tableMetadata == null || csvParser == null) {
+        if (tableMetadata == null) {
```

Comment:
Same as above comment. This line is marked as not necessary because of `csvParser.getHeaderNames()` but we should check if there is a problem too. Doesn't Eclipse mark it as well?

Reply:
After another check, `getCsvParser(...)` may return null in some cases

Reply:
fixed